### PR TITLE
LTD-3589: Consider Party flags when filtering Cases

### PR DESCRIPTION
**Pull request by saruniitr**
### Aim

The 'flags' filter doesn't consider flags related to parties hence not showing any Cases when filtered by any of these flags.

### CODE REVIEW

File: api/cases/managers.py [hunk 1]
```diff
@@ -110,19 +110,12 @@ def with_regime_entry(self, regime_entry):
     def with_flags(self, flags):
         case_flag_ids = self.filter(flags__id__in=flags).values_list("id", flat=True)
         org_flag_ids = self.filter(organisation__flags__id__in=flags).values_list("id", flat=True)
-        open_flag_ids = self.filter(
-            baseapplication__openapplication__application_countries__country__flags__id__in=flags
-        ).values_list("id", flat=True)
```
Comment (saruniitr) [inline api/cases/managers.py:115]: Removed as this and `goods_type` below not applicable for SIEL applications.

File: api/cases/tests/test_case_search.py [hunk 2]
```diff
@@ -394,6 +399,38 @@ def test_head_request_my_cases_tab(self):
...
+        response_data = response.json()["results"]
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(self.application_cases), len(response_data["cases"]))
+
+        for case in response_data["cases"]:
+            if flag_level == FlagLevels.GOOD:
+                self.assertIn(flag_id, [item["id"] for item in case["goods_flags"]])
+            if flag_level == FlagLevels.DESTINATION:
+                self.assertIn(flag_id, [item["id"] for item in case["destinations_flags"]])
```
Comment (kevincarrogan) [inline api/cases/tests/test_case_search.py:432]: These keys could also be part of the parameterized bit e.g.
```
    @parameterized.expand(
        [
            [FlagLevels.GOOD, FlagsEnum.SMALL_ARMS, "goods_flags"],
            [FlagLevels.DESTINATION, FlagsEnum.LU_COUNTER_REQUIRED, "destinations_flags"],
        ]
    )
```

**Review by kevincarrogan — APPROVED** [review]

File: api/cases/tests/test_case_search.py (hunk 2 above)
Comment (saruniitr) [reply api/cases/tests/test_case_search.py:432]: :+1:

File: api/cases/tests/test_case_search.py [hunk 3]
```diff
@@ -394,6 +399,35 @@ def test_head_request_my_cases_tab(self):
 
         self.assertEqual(response_headers["Resource-Count"], "2")
 
+    @parameterized.expand(
+        [
+            [FlagLevels.GOOD, FlagsEnum.SMALL_ARMS, "goods_flags"],
+            [FlagLevels.DESTINATION, FlagsEnum.LU_COUNTER_REQUIRED, "destinations_flags"],
+        ]
+    )
```
Comment (currycoder) [inline api/cases/tests/test_case_search.py:407]: It seems like `flag_level` is not used in this test?

**Review by currycoder — APPROVED** [review]
One question, otherwise looking good

File: api/cases/tests/test_case_search.py (hunk 3 above)
Comment (saruniitr) [reply api/cases/tests/test_case_search.py:407]: it was used before but after moving the key as a parameter, it is not used, will remove it now.

**Pull request merged**